drv/stm32xx-sys/src/main.rs

```diff
@@ -2,13 +2,51 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
-//! A driver for the STM32xx RCC and GPIO blocks, combined for compactness.
+//! A driver for the STM32xx RCC, GPIO, SYSCFG, and EXTI blocks, combined for
+//! compactness.
+//!
+//! # EXTI theory of operation
+//!
+//! `sys` serves as a multiplexer for pin-change interrupts generated by the
+//! EXTI unit. From a user perspective, this works as follows:
+//!
+//! 1. `sys` gets configured in the `app.toml` to route interrupts on a specific
+//!    pin to a specific task, using a specific notification.
+//! 2. That task configures the pin using `sys.gpio_configure` and friends
+//!    (probably as an input, possibly with a pull resistor).
+//! 3. That task _also_ configures the pin's edge sensitivity, using
+//!    `sys.gpio_irq_configure`. Instead of naming the pin in this API, the task
+//!    needs to pass the notification mask where it expects the IRQ to arrive.
```

Comment:
I think using notification masks _is_ the correct mechanism for this, but it does seem a little bit weird that this operation can potentially change the edge sensitivity for multiple pins, if they're mapped to the same notification bit. 

There's not, as I understand it, a way for a task to receive notifications from two pins in different edge sensitivities (and pullup configs) on the same notification bit. But, maybe that's fine...none of our tasks are anywhere near using up their 32 bits of notifications, so you can just use another one if you need different pin configs...

Reply:
Yeah, my general response (and I should probably put this in the comment) is: don't map multiple pins to a notification bit unless you want to treat them as 100% indistinguishable. Because it has that effect.

Reply:
yup, putting that in the comment works for me!